Content.Server/Body/Metabolism/MetabolizerComponent.cs

```diff
@@ -27,33 +30,44 @@ public class MetabolizerComponent : Component
         public float UpdateFrequency = 1.0f;
 
         /// <summary>
-        ///     From which solution will this metabolizer attempt to metabolize chemicals in its parent bodies' bloodstream,
-        ///     as opposed to a solution container on the metabolizing entity itself.
+        ///     From which solution will this metabolizer attempt to metabolize chemicals
         /// </summary>
         [DataField("solution")]
         public string SolutionName { get; set; } = SharedBloodstreamComponent.DefaultSolutionName;
 
         /// <summary>
-        ///     A dictionary mapping reagent string IDs to a list of effects & associated metabolism rate.
+        ///     Does this component use a solution on it's parent entity (the body) or itself
         /// </summary>
-        /// <returns></returns>
-        [DataField("metabolisms", required: true, customTypeSerializer:typeof(PrototypeIdDictionarySerializer<ReagentEffectsEntry, ReagentPrototype>))]
-        public Dictionary<string, ReagentEffectsEntry> Metabolisms = default!;
-    }
+        /// <remarks>
+        ///     Most things will use the parent entity (bloodstream).
+        /// </remarks>
+        [DataField("solutionOnBody")]
+        public bool SolutionOnBody = true;
 
-    [DataDefinition]
-    public class ReagentEffectsEntry
-    {
         /// <summary>
-        ///     Amount of reagent to metabolize, per metabolism cycle.
+        ///     List of metabolizer types that this organ is. ex. Human, Slime, Felinid, w/e.
         /// </summary>
-        [DataField("metabolismRate")]
-        public FixedPoint2 MetabolismRate = FixedPoint2.New(1.0f);
+        [DataField("metabolizerTypes", customTypeSerializer:typeof(PrototypeIdListSerializer<MetabolizerTypePrototype>))]
+        public List<string>? MetabolizerTypes = null;
```

Comment:
This should be a HashSet instead, the order shouldn't matter here.

Reply:
order doesn't matter